app/models/spree/shipment_decorator.rb

```diff
@@ -1,5 +1,12 @@
 Spree::Shipment.class_eval do
-  scope :exportable, -> { joins(:order).where('spree_shipments.state != ?', 'pending') }
+  scope :exportable, lambda {
```

Comment:
Yeah I like your refactor, I am more inclined to change it to a class method too.